### ISSUE TITLE
Error: getaddrinfo ENOTFOUND downloads.gradle-dn.com

### ISSUE THREAD

**Issue by jaredsburrows** [gradle/wrapper-validation-action#33]
Example:
https://github.com/jaredsburrows/android-gif-example/pull/133/checks?check_run_id=1419386504

Error:
```
Run gradle/wrapper-validation-action@v1
Error: getaddrinfo ENOTFOUND downloads.gradle-dn.com
```

**Comment by JLLeitschuh** [comment]
@jaredsburrows can you link to a specific line?

**Comment by jaredsburrows** [comment]
See https://github.com/jaredsburrows/android-gif-example/runs/1419386504?check_suite_focus=true#step:4:7.

**Comment by JLLeitschuh** [comment]
That looks like a DNS lookup failure. 'downloads.gradle-dn.com' is a DNS record configured through cloudflare. I wish we had a better stack trace here. This is probably happening just because the HTTP library that the GH action uses under the hood doesn't retry DNS lookups. Maybe we should do our own retrying in our code. @eskatos thoughts?

**Comment by JLLeitschuh** [comment]
@jaredsburrows is this the first time you've ever seen this?

**Comment by jaredsburrows** [comment]
Yes. I am wondering if it was some fluke. Like maybe there was not internet or something?

**Comment by JLLeitschuh** [comment]
Sounds like this is a transient DNS lookup failure issue. Simple fix is more robust retry, I think.

**Comment by eskatos** [comment]
I think that adding a retry wouldn't hurt given we only do idempotent requests.
Let's see if we get more reports before taking action.

**Comment by simonbasle** [comment]
not entirely the same, but I got `Error: connect ETIMEDOUT 104.17.160.20:443` today... that one might be less transient than a DNS lookup failure though, so not sure retry would have helped anyway

**Comment by JLLeitschuh** [comment]
Yea, so I think this is a sign we need to put in some retry logic.

**Comment by KengoTODA** [comment]
Hope that this issue has been resolved by #39

**Comment by JLLeitschuh** [comment]
Closed by #39
